### DISCUSSION TITLE
Using enums instead of strings

### DISCUSSION THREAD

**Discussion by drout0** [keras-team/keras#20651] [Ideas]
Please, could all strings in Keras be replaced by enums, which is the, may i say 'proper' way of representing symbols? 
besides being the 'proper' way, any error would be caught in the editor's syntax checker ( optimizer.sgb (error here)  instead of only being detected at runtime (optimizer='sgb' )
Ex. instead of model.compile(  optimizer='sgd', loss='binary_crossentropy',  metrics = ['accuracy', 'mae'] ):
    model.compile( [ optimizer.sgd, loss.binary_crossentropy, metrics.accuracy , metrics.mae, ... ],  ... )
then, inside model.compile, there would be a parser for the first argument (a list of enums) and the corresponding assignments based on the class of each enum (for e in enum_list:  if class(e) == optimizer: optimizer=e.value, etc etc
